dashboard/app/controllers/policy_compliance_controller.rb

```diff
@@ -18,13 +20,74 @@ def child_account_consent
     user = permission_request.user
     #Update User
     if user.child_account_compliance_state != User::ChildAccountCompliance::PERMISSION_GRANTED
-      user.child_account_compliance_state = User::ChildAccountCompliance::PERMISSION_GRANTED
-      user.child_account_compliance_state_last_updated = DateTime.now.new_offset(0)
+      user.update_child_account_compliance(User::ChildAccountCompliance::PERMISSION_GRANTED)
       user.save!
       parent_email = permission_request.parent_email
       ParentMailer.parent_permission_confirmation(parent_email).deliver_now
     end
     @permission_granted = true
     @permission_granted_date = user.child_account_compliance_state_last_updated
   end
+
+  # POST /policy_compliance/child_account_consent
+  # This URL creates a parental permission request. It is invoked via the
+  # lockout panel or potentially user settings by the child account.
+  def child_account_consent_request
+    # If we already comply, don't suddenly invalid it
+    if current_user.child_account_compliance_state == User::ChildAccountCompliance::PERMISSION_GRANTED
+      redirect_back fallback_location: lockout_path and return
+    end
+
+    # Only allow three unique permission requests per day
```

Comment:
It took me a while to grok this code. The key thing I was misunderstanding is that we are dealing with two limits:
1. There is a limit of 3 unique parent_emails per day.
2. There is a limit of 2 email resends per parent_email per lifetime.

I'm wondering if this overall description should be somewhere up near the top of method.

Reply:
I can do that. I'll add that to the docstring.

Reply:
Done.